File: 프로그래머스/unrated/147355. 크기가 작은 부분문자열/크기가 작은 부분문자열.js
function solution(t, p) {
    var answer = 0;
    let sbstring=new Array();
    
    
    for(let i=0;i<t.length-p.length+1;i++){
        
        if(Number(t.slice(i,i+p.length))<=Number(p)){
            
            sbstring.push(Number(t.slice(i,i+p.length)))
            
        }
        
    }
    
    answer=sbstring.length
    return answer;
}

File: 프로그래머스/1/147355. 크기가 작은 부분문자열/크기가 작은 부분문자열.java
class Solution {
    public int solution(String t, String p) {
        int answer = 0;
        Long b= Long.parseLong(p);
        for(int i=0 ;i <t.length()-p.length()+1;i++){
            if(t.length()==1){
                if(t.equals(p))return 1;
            }
            Long a = Long.parseLong(t.substring(i,i+p.length()));
            if(b>=a){
                answer++;
            }
            
        }
        
        return answer;
    }
}